src/snowflake/snowpark/session.py

```diff
@@ -952,15 +975,26 @@ def _resolve_packages(
                             ex,
                         )
 
-            if package_name in result_dict:
+            # Always allow Snowpark package to be overridden
+            if package_name in result_dict and package_name != SNOWPARK_PACKAGE_NAME:
                 if result_dict[package_name] != package:
                     raise ValueError(
-                        f"Cannot add {package} because {result_dict[package_name]} "
-                        "is already added"
+                        f"Cannot add package '{package}' because {result_dict[package_name]} "
+                        "is already added."
                     )
             else:
                 result_dict[package_name] = package
 
+        dependency_packages = None
+        if len(unsupported_packages) != 0:
+            _logger.warning(
+                f"The following packages are not available in Snowflake: {unsupported_packages}. They "
+                f"will be uploaded to session stage."
+            )
+            dependency_packages = self._upload_unsupported_packages(
```

Comment:
We should skip the auto-upload when it is in stored proc. Some of the steps are not supported in stored proc (e.g. `Popen`), also the unsupported packages are already uploaded from client, and the zip files are inherited from stored proc to its child jobs.

Reply:
The pip install `Popen` would occur on the client side; would that still affect the stored proc?

Reply:
Does skipping auto upload if inside stored proc only apply during testing? (Or do you mean that I need to add code here in `session.py` to skip)